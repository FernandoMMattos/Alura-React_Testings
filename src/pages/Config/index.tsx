import Footer from "src/components/Footer";
import Formulario from "src/components/Form";
import ListaParticipantes from "src/components/Lista";
import "../../index.css";

import Card from "src/components/Card";

const Config = () => {
  return (
    <Card>
      <section>
        <h2>Vamos comecar</h2>
        <Formulario />
        <ListaParticipantes />
        <Footer />
      </section>
    </Card>
  );
};

export default Config;
