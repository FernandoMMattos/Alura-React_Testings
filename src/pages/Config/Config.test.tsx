import { render } from "@testing-library/react";
import { RecoilRoot } from "recoil";
import Config from ".";

const mockNavegacao = jest.fn();

jest.mock("react-router-dom", () => {
  return {
    useNavigate: () => mockNavegacao,
  };
});

describe("a pagina de configuracao", () => {
  test("deve ser renderizada corretamente", () => {
    const { container } = render(
      <RecoilRoot>
        <Config />
      </RecoilRoot>
    );

    expect(container).toMatchSnapshot();
  });
});
