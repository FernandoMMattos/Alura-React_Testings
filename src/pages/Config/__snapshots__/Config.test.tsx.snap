// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a pagina de configuracao deve ser renderizada corretamente 1`] = `
<div>
  <div
    class="card"
  >
    <section>
      <h2>
        Vamos comecar
      </h2>
      <form>
        <div
          class="grupo_input_btn"
        >
          <input
            placeholder="Insira os nomes dos participantes"
            type="text"
            value=""
          />
          <button
            disabled=""
          >
            Adicionar
          </button>
        </div>
        
      </form>
      <ul />
      <footer
        class="rodape_configuracoes"
      >
        <button
          class="botao"
          disabled=""
        >
          Iniciar Brincadeira
        </button>
        <img
          alt="Sacolas de compras"
          src="/imagens/sacolas.png"
        />
      </footer>
    </section>
  </div>
</div>
`;
